### PR TITLE
Make Improvements to the Engineering Diffraction 2 GUI

### LINKED ISSUE #27189: Improve Existing Tabs in Engineering Diffraction 2 GUI
Based on user feedback some improvements should be made to the GUI:

- [x] Vanadium run files should also be saved to the user directories.
- [x] Focusing should output `.gss`, `.dat`, and `.nxs` files. 
- [x] The user should be provided with more status information, such as when the GUI is searching for files. 
- [x] Similar plots for multiple banks should be shown as tiled plots. 
- [x] Refactor current calibration into separate class.
- [x] Add Documentation for focusing tab
- [x] Make Ceria/Calib variables consistent.

### PR DESCRIPTION
**Pull request by cailafinn**
### ~Blocked By: #27486 #27589~

**Description of work.**
A few changes have been made to the code and functionality of the engineering diffraction GUI. 

-  Vanadium run files are also saved to the user directories.
- Focus outputs now include `.gss`, `.dat`, and `.nxs` files.
- The user is provided with more status information, such as when the GUI is searching for files.
- Similar plots for multiple banks are shown as tiled plots.
- Refactored current calibration into separate class.
- Added Documentation for focusing tab
- Made Ceria/Calib variables consistent, now using "`sample`"

**To test:**
1. Open Eng Diff GUI.
2. Click the calibrate button while the GUI is searching for files. 
2. Create a new calibration with "Plot Output" ticked (V: 307521 Sample: 305738)
3. Check similar Plots appear in the same figure. 
1. Check documentation is up to date. 
0. Go to the focus tab and focus a run (305761)
11. Check that all 3 file types have been generated. 

Fixes #27189 

*This does not require release notes* because **users have not had access to this GUI yet.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by Harrietbrown** [comment]
When focussing i am getting an error `omp_set_num_threads should only be called in serial regions`.
Should that be comp_set_num_threads?

File: scripts/Engineering/gui/engineering_diffraction/tabs/common/vanadium_corrections.py [hunk 1]
```diff
@@ -25,11 +25,12 @@
 SAVED_FILE_INTEG_SUFFIX = "_precalculated_vanadium_run_integration.nxs"
 
 
-def fetch_correction_workspaces(vanadium_path, instrument):
+def fetch_correction_workspaces(vanadium_path, instrument, rb_number=""):
     """
     Fetch workspaces from the file system or create new ones.
     :param vanadium_path: The path to the requested vanadium run raw data.
     :param instrument: The instrument the data came from.
+    :param rb_number:
```
Comment (Harrietbrown) [inline scripts/Engineering/gui/engineering_diffraction/tabs/common/vanadium_corrections.py]: rb_number should have a summery

File: scripts/Engineering/gui/engineering_diffraction/tabs/common/vanadium_corrections.py [hunk 2]
```diff
@@ -88,15 +99,19 @@ def _save_correction_files(integration_workspace, integration_path, curves_works
         return
 
 
-def _generate_saved_workspace_file_paths(vanadium_number):
+def _generate_saved_workspace_file_paths(vanadium_number, rb_num=""):
     """
     Generate file paths based on a vanadium run number.
     :param vanadium_number: The number of the vanadium run.
```
Comment (Harrietbrown) [inline scripts/Engineering/gui/engineering_diffraction/tabs/common/vanadium_corrections.py]: rb_num should be added to the paramiter list. Also for consistency you should have rb_number or rb_num in both functions

File: scripts/Engineering/gui/engineering_diffraction/tabs/focus/model.py [hunk 3]
```diff
@@ -75,17 +73,37 @@ def _plot_focused_workspace(focused):
         focused_wsp = Ads.retrieve(focused)
         plot([focused_wsp], wksp_indices=[0])
 
-    def _save_focused_output_files_as_nexus(self, instrument, sample_path, bank, sample_workspace,
-                                            rb_number):
+    def _save_output(self, instrument, sample_path, bank, sample_workspace, rb_number):
         """
-        Save a focused workspace to the file system. Saves a separate copy to a User directory if an rb number has been
-        set.
+        Save a focused workspace to the file system. Saves a separate copies to a User directory if an rb number has
```
Comment (Harrietbrown) [inline scripts/Engineering/gui/engineering_diffraction/tabs/focus/model.py]:
```suggestion
        Save a focused workspace to the file system. Saves separate copies to a User directory if an rb number has
```

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
I missed it before but along with the `omp_set_num_threads` error message there is also a `omp_set_dynamic should only be called in serial regions` message when focussing.

**Review by Harrietbrown — DISMISSED** [review]
Is the error message from `omp_set_num_threads` and `omp_set_dynamic` being handled in a different issue?

**Comment by ConorMFinn** [comment]
> Is the error message from `omp_set_num_threads` and `omp_set_dynamic` being handled in a different issue?

Hm, I don't get that message locally. So I thought it was solved after the rebase. I'll look into it.

**Comment by Harrietbrown** [comment]
did you get it when you were first running it?

**Comment by ConorMFinn** [comment]
I can't be sure, I did the rebase before looking at the suggestions. It isn't currently happening when I do `test-pr` either. So I'm guessing it's a windows thing.

**Comment by Harrietbrown** [comment]
I figured it would be if you hadn't noticed it before.

**Comment by ConorMFinn** [comment]
> 
> 
> I figured it would be if you hadn't noticed it before.

This issue appears to be caused by the `SaveGSS` algorithm. I can remove that funcionality, or leave it in and accept the error appearing until that algorithm is fixed. What do you think is the best solution?

**Comment by Harrietbrown** [comment]
In this case I would remove the functionality and put in a TODO, we don't want a bit error box that will mess up any other PRs you make untill you fix it. Best to fix it before it gets merged.

**Comment by ConorMFinn** [comment]
> 
> 
> In this case I would remove the functionality and put in a TODO, we don't want a bit error box that will mess up any other PRs you make untill you fix it. Best to fix it before it gets merged.

Issue for this created at #27582

**Review by Harrietbrown — APPROVED** [review]
Good that the saveGSS has been sorted out. Good job.

**Pull request merged**